packages/nuxt/src/module.ts

```diff
@@ -0,0 +1,151 @@
+import type { LoadClerkJsScriptOptions } from '@clerk/shared/loadClerkJsScript';
+import {
+  addComponent,
+  addImports,
+  addPlugin,
+  addServerHandler,
+  createResolver,
+  defineNuxtModule,
+  updateRuntimeConfig,
+} from '@nuxt/kit';
+
+export type ModuleOptions = Omit<LoadClerkJsScriptOptions, 'routerPush' | 'routerReplace'> & {
+  /**
+   * Skip the automatic server middleware registration. When enabled, you'll need to
+   * register the middleware manually in your application.
+   *
+   * @default false
+   *
+   * @example
+   *
+   * ```ts
+   * // server/middleware/clerk.ts
+   * import { clerkMiddleware } from '@clerk/nuxt/server'
+   *
+   * export default clerkMiddleware((event) => {
+   *   console.log('auth', event.context.auth)
+   * })
+   * ```
+   */
+  skipServerMiddleware?: boolean;
+};
+
+export default defineNuxtModule<ModuleOptions>({
+  meta: {
+    name: '@clerk/nuxt',
+    configKey: 'clerk',
+    compatibility: {
+      nuxt: '>=3.0.0',
```

Comment:
Do other integrations also define this compatibility in their `package.json#peerDependencies` or only in this key?

Reply:
Yeah. I double checked the modules available [here](https://github.com/nuxt-modules) and they all seem to have `compatibility.nuxt` instead of a peer dependency.

The [Nuxt module starter](https://github.com/nuxt/starter/tree/module) itself doesn't have a peer dep and relies to `compatibility.nuxt` as well

Reply:
Ok, good 👍 Then we can leave as-is